### ISSUE TITLE
Consider having channel relays conform to #NoBot and or #NoBridge mastodon etiquette/conventions

### ISSUE THREAD

**Issue by rectagonal** [friendica/friendica#14086]
Current etiquette on mastodon is for users to put the #NoBot and #NoBridge hashtags in their profile bio, if they do not want to their accounts to be monitored by automated bots or bridged to other networks respectively. While this of course is not a technical measure that prevents bots or bridges from disobeying the convention; it's considered polite to do so.

I recently had several toots in a row swept up by threadsrelay@sekretaerbaer.de in a way which raised my hackles, so to speak. It became obvious to me that a bot was boosting selected toots and I did not know why.

On the mastodon user side of this, the behavior feels very unusual, unexpected and potentially malicious unless your aware of the feature. In fact being largely unfamiliar with the last few years of friendica development, im still fuzzy on its use case; even though it has been explained to me.

I believe this threadsrelay@sekretaerbaer.de is what is called a channel relay in Friendica parlance, but if I got that wrong I apologize.

**Comment by annando** [comment]
Channel relays are automated accounts that reshare content based on content related rules, like for example used hashtags, keywords in the message, media or some other rules. They reshare in private, so that only subscribed accounts can see these reshares. The posts are not bridged to any other network, they stay in the Fediverse. Is there some documentation concerning `#NoBot` or `#NoBridge`?

**Comment by loma-one** [comment]
The #NoBot, #NoBridge discussion is centred on various 3Party products such as the Bluesky bridge, cvecrowd.com and other external services that access the AP stream. The entry in the bio is more of an emergency solution to defend against such services. In my opinion, something like this does not belong in the bio. It should rather be a profile setting. I don't know whether AP provides anything like this today.

**Comment by rectagonal** [comment]
> The #NoBot, #NoBridge discussion is centred on various 3Party products such as the Bluesky bridge, cvecrowd.com and other external services that access the AP stream.

So #NoBot definitely preceded most discussion of third party activity pub actors by a few years; the culture around it deffinitely hopes that most first party activitypub usage aspires to comply with it. I agree #NoBridge is recent, and perhaps I muddied the waters by even discussing that. You are right however, that CVECrowd does honor #NoBot today. https://kpwn.de/2023/12/cve-crowd-conditions-for-posts-to-be-listed/

> . In my opinion, something like this does not belong in the bio. It should rather be a profile setting. I don't know whether AP provides anything like this today.

I believe everyone using the hashtag today agrees with that. However, despite such interest mainline Mastodon has never accepted or implemented such a feature.

I will add that doing nothing, does seem like it will cause a number of well intentioned channel relays to be reported to mastodon moderators as bots not respecting the #NoBot hashtag. Which is probably undesirable. There are instances whose moderation practices will block the relay channel from all users. Where as the alternative is to allow some to opt out.Thus living with the friction effectively limits the relay channel further than neccessarry.

**Comment by rectagonal** [comment]
> They reshare in private, so that only subscribed accounts can see these reshares.

The account being reshared however, is notified of the reshare. Which is why from many mastodon users point of view, it becomes so obvious so quickly an unknown bot is monitoring select toots.



> Is there some documentation concerning `#NoBot` or `#NoBridge`?

It being more of a convention that a feature of the platform, I am honestly unsure on where to point you. The description in this blog post around it's usage in CVECrowd does explain #NoBot well. https://kpwn.de/2023/12/cve-crowd-conditions-for-posts-to-be-listed/

**Comment by annando** [comment]
By now we already check for public posts, so that already unlisted posts are not reshared. What I can add without many problems is that also posts are only reshared, when the profile of the owner of that post isn't set to "unlisted".

Also consider that the account only sends the "reshare" without any content. So even when such a message was sent to a system that is blocked by the original server, then that system had to try to fetch the original item and would fail.

Concerning the uncertainty of why an account reshared a post, I will recommend to the owners of these relay accounts to add a meaningful description with a responsible person that can be contacted.

**Comment by sekretaerbaer** [comment]
As I am the owner of the relay, I have a question of understanding. If I block a person via the relay account, does this mean that the underlying channel will no longer share the person's posts or do I have to set this in the channel settings? 
If I have to do this in the channel settings, then how?

**Comment by annando** [comment]
I have to investigate about the best method. You can define it in the search text, when you already use it. When you only use hash tags, I will have to check some stuff before.

**Comment by sekretaerbaer** [comment]
at the moment the relay only uses hashtags, but it should not be a problem to relalise this via the full text search with tag:tagname . am I right in assuming that it is enough if I add a "-from:nickname@domain.tld to the search" ?

**Comment by loma-one** [comment]
@rectagonal It is what it is ;)

In the end it remains an undocumented and therefore unknown gentlemen's agreement in the Fediverse. 

Looking to the future: 
it needs is a setting similar to the "do not track" for the two hashtags that can be sent via AP, to which all integrators of AP can orientate themselves.

**Comment by annando** [comment]
@sekretaerbaer I'm not totally sure, if adding this single definition helps or if you need some additional statement there.

**Comment by sekretaerbaer** [comment]
@annando I have just tested it with two accounts from different instances and it seems to work.

**Comment by MrPetovan** [comment]
@annando I think we should add at least a profile flag that is set (or unset) on contact update and based on the presence of `#nobot` and `#nobridge` in profile description that relays (not just channel relays) should check before sharing a post by these profiles. It's definitely more work, but the introduction of channel relays makes it much easier for Friendica users to set up automated accounts that can rub people across the Fediverse the wrong way. Especially since we also bridge protocols natively, although I'm not sure Diaspora users would be able to see reshared ActivityPub content.

**Comment by annando** [comment]
These posts are not shared to Diaspora, since we only perform the `Announce` activity. BTW: What other methods do you mean with "not just channel relays"?

**Comment by MrPetovan** [comment]
Isn't Friendica able to work as a regular relay, without a channel filter?

**Comment by annando** [comment]
Yeah, there is the "direct relay" mechanism. But I don't really see a reason to change anything there. This is just distributing some posts from local users to a broader audience.

**Comment by MrPetovan** [comment]
Fair enough, I still think we should honor the `#nobot` hashtag in profile description.
